File: TDDMicroExercises.ProposedSolution/CSharp/UnicodeFileToHtmTextConverter/UnicodeFileToHtmTextConverter.cs
using System.IO;
using System.Web;

namespace TDDMicroExercises.OneSolution.UnicodeFileToHtmTextConverter
{
    public class UnicodeFileToHtmTextConverter
    {
        private IUnicodeTextSource _textSource;

        public UnicodeFileToHtmTextConverter(string fullFilenameWithPath)
            : this (new UnicodeTextSourceFromFile(fullFilenameWithPath))
        {
        }

        public UnicodeFileToHtmTextConverter(IUnicodeTextSource textSource)
        {
            _textSource = textSource;
        }

        public string ConvertToHtml()
        {
            using (TextReader unicodeFileStream = _textSource.GetTextReader())
            {
                string html = string.Empty;

                string line = unicodeFileStream.ReadLine();
                while (line != null)
                {
                    html += HttpUtility.HtmlEncode(line);
                    html += "<br />";
                    line = unicodeFileStream.ReadLine();
                }

                return html;
            }
        }
    }
}

File: TDDMicroExercises/Java/TextConverter/src/main/java/tddmicroexercises/textconvertor/UnicodeFileToHtmTextConverter.java
package tddmicroexercises.textconvertor;

import java.io.File;
import java.io.FileReader;

public class UnicodeFileToHtmTextConverter
{
    private String _fullFilenameWithPath;


    public UnicodeFileToHtmTextConverter(String fullFilenameWithPath)
    {
        _fullFilenameWithPath = fullFilenameWithPath;
    }

    public String ConvertToHtml(){
    
    File file = new File(_fullFilenameWithPath);
    FileReader filereader = new FileReader(file);
    
    filereader.
    
    {
        using (TextReader unicodeFileStream = File.OpenText(_fullFilenameWithPath))
        {
            String html = "";

            String line = unicodeFileStream.ReadLine();
            while (line != null)
            {
                html += HttpUtility.HtmlEncode(line);
                html += "<br />";
                line = unicodeFileStream.ReadLine();
            }

            return html;
        }
    }
    }
}
